plugins/inputs/tail/tail_test.go

```diff
@@ -1030,3 +1030,172 @@ func TestInitInitialReadOffset(t *testing.T) {
 		})
 	}
 }
+
+// TestTailNoLeak tests that we don't leak file descriptors when repeatedly
+// tailing the same file across multiple Gather calls
+func TestTailNoLeak(t *testing.T) {
+	// Create a temp directory for our test file
+	tempDir := t.TempDir()
+	logFile := filepath.Join(tempDir, "test.log")
+
+	content := "cpu usage_idle=100\r\n"
+	require.NoError(t, os.WriteFile(logFile, []byte(content), 0600))
+
+	// Setup the plugin
+	tt := newTestTail()
+	tt.Log = testutil.Logger{}
+	tt.InitialReadOffset = "beginning"
+	tt.Files = []string{logFile}
+	tt.SetParserFunc(newInfluxParser)
+	require.NoError(t, tt.Init())
+
+	// Start the plugin
+	var acc testutil.Accumulator
+	require.NoError(t, tt.Start(&acc))
+	defer tt.Stop()
+
+	// Wait for the plugin to process the file
+	acc.Wait(1)
+
+	// Make sure we got the first metric
+	acc.AssertContainsFields(t, "cpu",
+		map[string]interface{}{
+			"usage_idle": float64(100),
+		})
+
+	// Call Gather multiple times to simulate multiple collection intervals
+	require.NoError(t, acc.GatherError(tt.Gather))
+	require.NoError(t, acc.GatherError(tt.Gather))
+	require.NoError(t, acc.GatherError(tt.Gather))
+
+	// Verify we only have one tailer for the file, not multiple
+	require.Len(t, tt.tailers, 1, "Expected only one tailer despite multiple Gather calls")
```

Comment:
Isn't this racy as the plugin is still running and might pick up new tailers?

Reply:
Thank you for catching that potential race condition.